sg_covid_impact/make_sic_division.py

```diff
@@ -1,128 +1,82 @@
+# %%
+"""Script to aggregate NOMIS data over SIC divisions."""
 import os
-import re
 import logging
+
 import pandas as pd
-from sg_covid_impact.getters.nomis import get_BRES, get_IDBR
-from sg_covid_impact.sic import load_sic_taxonomy, extract_sic_code_description
 
 import sg_covid_impact
+from sg_covid_impact.getters.nomis import get_BRES, get_IDBR
+from sg_covid_impact.sic import (
+    load_sic_taxonomy,
+    extract_sic_code_description,
+    save_sic_taxonomy,
+    _SIC_OUTPUT_FILE,
+)
+
 
 project_dir = sg_covid_impact.project_dir
 
 
-def make_sic_target_lookup(sic, source_level="Class", target_level="Division"):
-    """Makes a lookup between source SIC level (eg class ie SIC-4)
-        and a target level (eg Division)
-    Args:
-        sic (table): the sic taxonomy
-        source_level (str): the sic level we are converting from
-        target_level: the sic level we want to convert into
-    Returns:
-        A dict mapping source levels into target levels
-    """
-    lookup_table = (  # Each row in this lookup has a source and a target
-        sic[[target_level, source_level]].fillna(method="ffill").dropna(axis=0)
+def make_sic_div_lookup(sic):
+    """Creates lookup between SIC class and division."""
+    return (
+        sic.assign(
+            Division=lambda x: x.Class.str[0:2],
+            Class=lambda x: x.Class.str.replace(".", ""),
+        )
+        .set_index("Class")["Division"]
+        .to_dict()
     )
-    # Remove periods from the sic levels
-    lookup_table[source_level] = lookup_table[source_level].apply(
-        lambda x: re.sub("\\.", "", x)
-    )
-
-    return lookup_table.set_index(source_level)[target_level].to_dict()
 
 
-def convert_sic_data(
+def aggregate_NOMIS_over_SIC(
     table,
-    source_target_lookup,
-    code_descr_lookup,
-    name,
-    source_level="SIC4",
-    target_level="division",
+    target_level,
 ):
-    """Converts econ activity table from source level (SIC) to target_level
-    Args:
-        table (pd.DataFrame): official data
-        source_target_lookup (dict): lookup between source codes (SIC) and target codes
-        code_descr_lookup (dict): lookup between target codes and
-            descriptions
-        name (str): name for the data source
-        source_level (str): name for the source level we are converting from
-        target_level (str): target sic level we are converting into
-    Returns a table with employment or establishment data by target sic level
-        and geography
-    """
-    table_ = table.copy()
+    """Aggregates NOMIS table from SIC4 (Class) to `target_level`.
 
-    # Extract division from SIC4s
-    table_[target_level] = table_[source_level].map(source_target_lookup)
```

Comment:
This can happen at a higher-level in the code (then we don't nave to pass `source_target_lookup` down two levels of functions)